### PR TITLE
Fix the openshift docker file build command.

### PR DESCRIPTION
**Pull request by fedepaol**
Also, vendor the gingko executable so it can be used without downloading deps.

### CODE REVIEW

File: hack/build-test-bin.sh [hunk 1]
```diff
@@ -8,13 +8,15 @@ if ! which go; then
 fi
 
 GOPATH="${GOPATH:-~/go}"
+GOFLAGS="${GOFLAGS:-"-mod=vendor"}"
+
 export PATH=$PATH:$GOPATH/bin
 
 if ! which gingko; then
```
Comment (SchSeba) [inline hack/build-test-bin.sh:15]: now that we use the tools package can you change this to build it from the vendor folder?

File: cnf-tests/Dockerfile.openshift [hunk 2]
```diff
@@ -9,7 +9,7 @@ RUN mkdir -p $PKG_PATH
 COPY . $PKG_PATH/
 WORKDIR $PKG_PATH
 
-RUN make build-bin
+RUN make test-bin
 
 WORKDIR /
```
Comment (SchSeba) [inline cnf-tests/Dockerfile.openshift:14]: why we have this line here?

File: hack/build-test-bin.sh (hunk 1 above)
Comment (fedepaol) [reply hack/build-test-bin.sh:15]: it works by having GOFLAGS=-mod=vendor.
The issue is that in the openshift base image GOFLAGS is already set like that, hence line 12.
With that variable, all build happen using the fendor folder.

File: cnf-tests/Dockerfile.openshift (hunk 2 above)
Comment (fedepaol) [reply cnf-tests/Dockerfile.openshift:14]: You must really hate this. You made the same question before :-)
https://github.com/openshift-kni/cnf-features-deploy/pull/138#discussion_r409425961

I took inspiration from other dockerfiles, I guess it's to restore WORKDIR to what it was before.

Comment (SchSeba) [reply cnf-tests/Dockerfile.openshift:14]: yeah but then we jump to another container this is only the builder container

File: hack/build-test-bin.sh (hunk 1 above)
Comment (SchSeba) [reply hack/build-test-bin.sh:15]: great thanks!

File: cnf-tests/Dockerfile.openshift (hunk 2 above)
Comment (fedepaol) [reply cnf-tests/Dockerfile.openshift]: removed it

**Comment by fedepaol** [comment]
Validated the dockerfile locally, CI is broken.
Overriding it.

**Pull request merged**